audit-checklists/Arbitrum.md

```diff
@@ -0,0 +1,95 @@
+
+> Make sure the account provided as `callValueRefundAddress` cannot misuse its capability to cancel the ticket in a way that hurts the protocol
+
+Bug examples: [1](https://github.com/code-423n4/2024-05-olas-findings/issues/29)
+
+## Oracle integration
+
+### Sequencer uptime
+
+Typically users interact with Arbitrum through sequencer. Sequencer is responsible for including and processing transactions. If sequencer goes down users can still submit transaction using [Inbox](https://etherscan.io/address/0x4Dbd4fc535Ac27206064B68FfCf827b0A60BAB3f) contract on L1. However, such transactions will be included with potentially significant delay. In the worst case, if sequencer is down for a prolonged time, transactions submitted to L1 can be force included only after 24h. In the context of price feeds, it means that reported prices can get stale and significantally drift from the actual price. Thus protocols should always check the sequencer's uptime and revert the price request if sequencer is down for more than predefined period of time.
+
+Chainlink's sequencer uptime feeds can be used for this purpose. Integration example can be found [here](https://docs.chain.link/data-feeds/l2-sequencer-feeds#example-code).
+
+> Integrations with Chainlink's price feeds on Arbitrum should always check the sequencer uptime in order to avoid stale prices
+
+Bug examples: [1](https://code4rena.com/reports/2024-07-benddao#m-12--no-check-if-arbitrumoptimism-l2-sequencer-is-down-in-chainlink-feeds-priceoraclesol-), [2](https://github.com/shieldify-security/audits-portfolio-md/blob/main/Pear-v2-Security-Review.md#m-02-missing-check-for-active-l2-sequencer-in-calculatearbamount)
+
+### Incorrect staleness threshold or feed decimals used
+
+Staleness threshold is used to determine if price fetched from price feed is fresh enough. Most often staleness threshold should match the feed's heartbeat - the longest amount of time that can pass between consequent price updates. Also, feeds use different number of decimals (not necessarily related to number of decimals used by underlying tokens). Using incorrect values for staleness or feed decimals can lead to critical outcomes. Different feeds on Arbitrum require different values for mentioned parameters. For example:
+
+- [LINK / ETH](https://arbiscan.io/address/0xb7c8Fb1dB45007F98A68Da0588e1AA524C317f27) has 24h heartbeat and uses 18 decimals
+- [LINK / USD](https://arbiscan.io/address/0x86E53CF1B870786351Da77A57575e79CB55812CB) has 1h heartbeat and uses 8 decimals
+
+> Make sure that contract uses correct staleness threshold and decimal precision for every price feed being integrated
+
+Bug examples: [1](https://github.com/sherlock-audit/2024-06-new-scope-judging/issues/166), [2](https://github.com/sherlock-audit/2024-06-new-scope-judging/issues/9),
+[3](https://github.com/solodit/solodit_content/blob/main/reports/Zokyo/2024-06-23-Copra.md#incorrect-staleness-threshold-for-chainlink-price-feeds)
```

Comment:
If you're putting a link to some repository don't point to main but to a specific commit --https://github.com/solodit/solodit_content/blob/d4d5dbc/reports/Zokyo/2024-06-23-Copra.md#incorrect-staleness-threshold-for-chainlink-price-feeds